analyzer/src/main/kotlin/managers/GoMod.kt

```diff
@@ -515,6 +502,19 @@ private data class ModuleInfoFile(
     )
 }
 
+/**
```

Comment:
> Make the `Graph` agnostic of `Go`

Can you explain this a bit further in the commit message? At a first glance, there seems to be nothing Go-specific in that function, and the word "Go" isn't mentioned. My guess is that it's the *assumption* being made in the function that's Go-specific, i.e. that there's only one package without a version, and that that's the main package.

Reply:
And even if the code is Go-specific, as the whole Graph still is Go-specific (Only used in Go-context, and that will probably not change, or?), why not just leave it there?

Reply:
> why not just leave it there?

The file is very large. I find it easier to work with the code if this is in a separate file.

Reply:
> If you mean the function: To me the logic belong more closely to `GoMod` than to that `Graph`. IMO a `Graph` is something everybody knows what it is about, if there are no built-in surprises like this.

Yes, I meant the function. As long as `Graph` is not useful / being used in another context than Go, I don't see a reason to extract it. But now that it already is, let's leave it like that.